var/spack/repos/builtin/packages/chatterbug/package.py

```diff
@@ -0,0 +1,89 @@
+##############################################################################
+
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Chatterbug(MakefilePackage):
+    """A suite of communication-intensive proxy applications that mimic
+       commonly found communication patterns in HPC codes. These codes can be
+       used as synthetic codes for benchmarking, or for trace generation using
+       OTF2.
+    """
+    tags = ['proxy-app']
+
+    homepage = "https://chatterbug.readthedocs.io"
+    git      = "https://github.com/LLNL/chatterbug.git"
+
+    version('develop', branch='master')
+    version('1.0', tag='v1.0')
+
+    variant('otf2', default=False, description='Build with OTF2 tracing')
+
+    depends_on('mpi')
+    depends_on('scorep~gui', when='+otf2')
+
+    @property
+    def build_targets(self):
+        targets = []
+
+        cxxflag = ' -O3 '
+
+        if '+otf2' in self.spec:
+            cxxflag += '-I{0}'.format(spec['scorep'].prefix.include)
```

Comment:
This information should ideally be available through ‘headers()’ of scorep package.

Reply:
I just did some search on other packages. Are you suggesting that the scorep package should define headers() that calls find_headers() on the two locations?

Reply:
yes, look at `hypre` or `petsc` for simple examples.  Here you would do `spec['scorep'].headers.include_flags`

Reply:
Unless compiler wrappers aren’t being used, you shouldn’t need to add includes or library search paths for dependencies. They’re already injected by the compiler wrappers.  It looks like this build does use them.

Reply:
@tgamblin 

> Unless compiler wrappers aren’t being used

i think they won't work here as AFAIR they do not rely on `headers()` but instead hardcode `prefix.include`, whereas the next line here adds `.prefix.include.scorep`. 

p.s. compiler wrappers and rpaths should use more `libs()` and `headers()` to handle non-trivial cases like `intel` and packages here.

Reply:
Adding results of libs and headers to the build environment wouldn’t be so hard... that’s something we should do.  They just need to be added to the env cars for the wrappers.